src/v/config/property.h

```diff
@@ -566,6 +593,69 @@ class conversion_binding : public binding_base<T> {
     }
 };
 
+/**
+ * A property binding contains a copy of the property's value, which
+ * will be updated in-place whenever the property is updated in the
+ * cluster configuration.
+ *
+ * This is useful for classes that want a copy of a property without
+ * having to write their own logic for subscribing to value changes.
+ */
+template<class T>
+class sanctioning_binding : public binding_base<T> {
+private:
+    T _value;
+    T _default_value;
+    bool _is_value_restricted;
+
+    bool do_update(const T& v, bool is_restricted) override {
+        auto changed = _value != v;
+        _value = v;
+        _is_value_restricted = is_restricted;
+        return changed;
+    }
```

Comment:
I think it would be cleaner if we only allowed creating `sanctioning_binding`'s on `enterprise<property<T>>`'s. We wouldn't need to propagate the `_is_restricted` param everywhere, and instead, we could just query the enterprise property in this method:
```suggestion
    bool do_update(const T& v) override {
        auto changed = _value != v;
        _value = v;
        _is_value_restricted = _enterprise_parent.check_restricted(_value);
        return changed;
    }
```

Reply:
I would like this, but all the parent logic is currently in the `binding_base`. All the derived bindings don't know anything about the parent. While it is limiting, i find it quite neat as a design, so i didn't want to change this part.
Following the existing design, in the derived binding we can only use what the `binding_base` is forwarding.

Reply:
i am +1 on what Gellert proposed